shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/ManagedProcessor.java

```diff
@@ -0,0 +1,189 @@
+package com.redhat.service.smartevents.shard.operator.v2.resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ProcessorDTO;
+import com.redhat.service.smartevents.shard.operator.core.utils.LabelsBuilder;
+import com.redhat.service.smartevents.shard.operator.core.utils.StringUtils;
+import io.fabric8.kubernetes.api.model.Namespaced;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.utils.KubernetesResourceUtil;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.ShortNames;
+import io.fabric8.kubernetes.model.annotation.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Objects.requireNonNull;
+
+@Group("com.redhat.service.smartevents")
+@Version("v2alpha1")
+@ShortNames("mproc")
+public class ManagedProcessor extends CustomResource<ManagedProcessorSpec, ManagedProcessorStatus> implements Namespaced {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessor.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public static final String COMPONENT_NAME = "executor";
+    public static final String OB_RESOURCE_NAME_PREFIX = "ob-";
+
+    /**
+     * Don't use this default constructor!
+     * This class should have a private default constructor. Unfortunately, it's a CR which is created via reflection by fabric8.
+     * <p/>
+     * Use {@link #fromBuilder()} to create new instances.
+     */
+    public ManagedProcessor() {
+        this.setStatus(new ManagedProcessorStatus());
+    }
+
+    /**
+     * Standard way of creating a new {@link ManagedProcessor}.
+     * This class has a public constructor for integration with Kubernetes libraries only.
+     * Please don't use the public constructor to create references of this CR.
+     *
+     * @return a Builder to help client code to create new instances of the CR.
+     */
+    public static Builder fromBuilder() {
+        return new Builder();
+    }
+
+    public static ManagedProcessor fromDTO(ProcessorDTO processorDTO, String namespace, String executorImage) {
```

Comment:
We don't need the executorImage field any more. Camel-K generates the image for us as part of the build.